pilot/pkg/model/push_context.go

```diff
@@ -134,6 +135,26 @@ type PushContext struct {
 	networkGateways map[string][]*Gateway
 }
 
+// BetaPolicy placeholder to return betap policy for mTLS.
+// Note: for reviewer, this will be updated to a cache
+// map, k = <namespace, has(label-set), port>, value : true or false
+// determing whether this particular port can accept mTLS.
+// In future, authn applier can also benefits from the same optimization.
+func (ps *PushContext) EndpointAcceptMtls(
+	namespace string, labels labels.Collection, port uint32) bool {
```

Comment:
Do you really need to go down to the labels for now? Why not simply look up by the endpoint's namespace and decide, just like the way it was done in v1alpha1 ?

Reply:
If mTLS is disabled for this label (via peer authn), then we don't want to use mTLS to that endpoint.
Btw, we've never looked at endpoint namespace, right?

Reply:
this is due to the beta policy's semantics... for example

1. I create a mesh wide beta policy to disable mTLS. everything by default is plaintext
1. i have two deployments:
  1. prod deployment with label { app: shopping-frontend, env: production }
   2.staging deployment with label { app: shopping-frontend, env: staging}
  one shopping-frontend using ` { app: shopping-frontend } `
1. I configure a beta policy using workload selector to enable mTLS 
   ```yaml
   mode: STRICT
   selector:
      env: production
   ```

So the clients of `shpping-frontend.svc.cluster.local` can send traffic to both endpoint from deployments, but only prod deployment can serve mTLS due to the policy configures.

Reply:
The convenience part of the v1 policy is because it's either namespace or service. and one outbound cluster corresponds to 1 service. mTLS mode enum is materialistically known at the ounbound cluster/service level.